src/Refitter.Core/RefitInterfaceGenerator.cs

```diff
@@ -193,11 +196,17 @@ protected static bool IsApiResponseType(string typeName)
 
     private string GetConfiguredReturnType(string returnTypeParameter)
     {
+        var asyncType = GetAsyncOperationType(false);
         return settings.ReturnIApiResponse
-            ? $"Task<IApiResponse<{WellKnownNamesspaces.TrimImportedNamespaces(returnTypeParameter)}>>"
-            : $"Task<{WellKnownNamesspaces.TrimImportedNamespaces(returnTypeParameter)}>";
+            ? $"{asyncType}<IApiResponse<{WellKnownNamesspaces.TrimImportedNamespaces(returnTypeParameter)}>>"
+            : $"{asyncType}<{WellKnownNamesspaces.TrimImportedNamespaces(returnTypeParameter)}>";
     }
 
+    private string GetAsyncOperationType(bool withVoidReturnType) =>
+        settings.ReturnIObservable
+            ? "IObservable" + (withVoidReturnType ? "<Unit>" : string.Empty)
```

Comment:
I don't have much experience working with `IObservable<T>` but why do we need `Unit`? What does it do and what is it for?

The examples in the [Refit README](https://github.com/reactiveui/refit?tab=readme-ov-file#retrieving-the-response) uses an `HttpResponseMessage` together with `IObservable`

```csharp
// Returns the raw response, as an IObservable that can be used with the
// Reactive Extensions
[Get("/users/{user}")]
IObservable<HttpResponseMessage> GetUser(string user);
```